Alpine compilation error pioneer 20190203.

My pioneer version (and  OS):

20190203 Alpinelinux

I have a stacking error in 97% with the file OSPosix.cpp Here the error

/home/diego/Descargas/FPS/games/New/pioneer-20190203/src/posix/OSPosix.cpp: In function 'void OS::EnableFPE()':
/home/diego/Descargas/FPS/games/New/pioneer-20190203/src/posix/OSPosix.cpp:55:3: error: 'feenableexcept' was not declared in this scope
   feenableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
   ^~~~~~~~~~~~~~
/home/diego/Descargas/FPS/games/New/pioneer-20190203/src/posix/OSPosix.cpp:55:3: note: suggested alternative: 'feraiseexcept'
   feenableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
   ^~~~~~~~~~~~~~
   feraiseexcept
/home/diego/Descargas/FPS/games/New/pioneer-20190203/src/posix/OSPosix.cpp: In function 'void OS::DisableFPE()':
/home/diego/Descargas/FPS/games/New/pioneer-20190203/src/posix/OSPosix.cpp:62:3: error: 'fedisableexcept' was not declared in this scope
   fedisableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
   ^~~~~~~~~~~~~~~
/home/diego/Descargas/FPS/games/New/pioneer-20190203/src/posix/OSPosix.cpp:62:3: note: suggested alternative: 'feraiseexcept'
   fedisableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
   ^~~~~~~~~~~~~~~
   feraiseexcept
make[2]: *** [CMakeFiles/pioneerLib.dir/build.make:4990: CMakeFiles/pioneerLib.dir/src/posix/OSPosix.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:192: CMakeFiles/pioneerLib.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,14 @@ if (CMAKE_COMPILER_IS_GNUCXX)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive")
 endif (CMAKE_COMPILER_IS_GNUCXX)
 
+include(CheckSymbolExists)
+check_symbol_exists(feclearexcept "fenv.h" HAS_FECLEAREXCEPT)
+check_symbol_exists(feenableexcept "fenv.h" HAS_FEENABLEEXCEPT)
+check_symbol_exists(fedisableexcept "fenv.h" HAS_FEDISABLEEXCEPT)
+if (HAS_FECLEAREXCEPT AND HAS_FEENABLEEXCEPT AND HAS_FEDISABLEEXCEPT)
+	set(HAS_FPE_OPS ON)
+endif()
+
 if (NOT CMAKE_BUILD_TYPE)
 	set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING
 		"Choose the type of build, options are: None(CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) Debug Release RelWithDebInfo MinSizeRel."

--- a/buildopts.h.cmakein
+++ b/buildopts.h.cmakein
@@ -7,5 +7,6 @@
 
 #cmakedefine01 WITH_OBJECTVIEWER
 #cmakedefine01 WITH_DEVKEYS
+#cmakedefine01 HAS_FPE_OPS
 
 #endif /* BUILDOPTS_H */

--- a/src/posix/OSPosix.cpp
+++ b/src/posix/OSPosix.cpp
@@ -3,6 +3,7 @@
 
 #include "FileSystem.h"
 #include "OS.h"
+#include "buildopts.h"
 #include <SDL.h>
 #include <fenv.h>
 #include <sys/time.h>
@@ -48,7 +49,7 @@ namespace OS {
 
 	void EnableFPE()
 	{
-#if defined(_GNU_SOURCE) && !defined(__APPLE__)
+#if HAS_FPE_OPS
 		// clear any outstanding exceptions before enabling, otherwise they'll
 		// trip immediately
 		feclearexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
@@ -58,7 +59,7 @@ namespace OS {
 
 	void DisableFPE()
 	{
-#if defined(_GNU_SOURCE) && !defined(__APPLE__)
+#if HAS_FPE_OPS
 		fedisableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
 #endif
 	}

